### PR TITLE
[Docs][API] Reorganize Sylius API docs + add authorization in Swagger UI

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

screencapture-localhost-63342-Sylius-docs-build-api-index-html-2020-03-25-08_44_11

### CODE REVIEW

File: docs/api/index.rst [hunk 1]
```diff
@@ -1,38 +1,73 @@
...
+==============
+
+.. warning::
+
+    The new, unified Sylius API is still under development.
+
+We have decided that we should rebuild our API and use API Platform to build a truly mature, multi-purpose API
+which can define a new standard for headless e-commerce backends.
+
+We will be supporting API Platform out-of-the-box. Secondly, it means that both APIs will be deprecated. We are not
```
Comment (CoderMaggie) [inline docs/api/index.rst]:
```suggestion
We will be supporting API Platform out-of-the-box. Secondly, it means that both APIs (Admin API and Shop API) will be deprecated. We are not
```

File: docs/api/index.rst [hunk 2]
```diff
@@ -1,38 +1,73 @@
...
+dropping them right now, but they will not receive further development. In the later phase, we should provide
+an upgrade path for currently working apps. Last, but not least, you can already tracktrace our progress. All the PR’s
+will be aggregated `in this issue <https://github.com/Sylius/Sylius/issues/11250>`_ and the documentation can be already found
+`here <http://master.demo.sylius.com/new-api/docs>`_.
+
+Authorization
+-------------
+
+All admin routes are protected by JWT authentication. If you would like to test these endpoints
```
Comment (CoderMaggie) [inline docs/api/index.rst]:
```suggestion
In the new API all admin routes are protected by JWT authentication. If you would like to test these endpoints
```

File: docs/api/index.rst [hunk 3]
```diff
@@ -1,38 +1,73 @@
...
+`in our Swagger UI docs <http://master.demo.sylius.com/new-api/docs>`_, you need to retrieve a JWT token firstly.
```
Comment (CoderMaggie) [inline docs/api/index.rst]:
```suggestion
`in our Swagger UI docs <http://master.demo.sylius.com/new-api/docs>`_, you need to retrieve a JWT token first.
```

File: docs/api/index.rst [hunk 4]
```diff
@@ -1,38 +1,73 @@
...
+You could do that by using an endpoint with default credentials for API administrators:
+
+.. image:: ../_images/sylius_api/api_platform_authentication_endpoint.png
+    :align: center
+    :scale: 50%
+
+|
+
+In the response you will get a token that need to be passed in each request header. In the Swagger UI, you can set
```
Comment (CoderMaggie) [inline docs/api/index.rst]:
```suggestion
In the response, you will get a token that has to be passed in each request header. In the Swagger UI, you can set
```

File: docs/api/index.rst [hunk 5]
```diff
@@ -1,38 +1,73 @@
...
+.. warning::
+
+    The new, unified Sylius API is still under development.
+
+We have decided that we should rebuild our API and use API Platform to build a truly mature, multi-purpose API
+which can define a new standard for headless e-commerce backends.
+
+We will be supporting API Platform out-of-the-box. Secondly, it means that both APIs (Admin API and Shop API) will
+be deprecated. We are not dropping them right now, but they will not receive further development. In the later phase,
+we should provide an upgrade path for currently working apps. Last, but not least, you can already tracktrace our progress.
```
Comment (lchrusciel) [inline docs/api/index.rst]:
```suggestion
we should provide an upgrade path for currently working apps. Last, but not least, you can already track our progress.
```

**Comment by lchrusciel** [comment]
Thanks, Grzegorz! :tada:

**Pull request merged**